src/kedro_databricks/cli/bundle/override_resources.py

```diff
@@ -138,14 +178,263 @@ def _get_old_value(result: Any, key: Any, value: Any):
     return result.get(key, default)
 
 
+def _split_updates_by_type(
+    updates: list[dict[str, Any]], lookup_key: str
+) -> tuple[
+    list[tuple[str, dict[str, Any]]], list[tuple[re.Pattern[str], dict[str, Any], str]]
+]:
+    """Split update items into literal and regex-based entries while preserving order.
+
+    The function identifies regex-based entries by the ``re:`` prefix on the
+    ``lookup_key`` value and compiles them for later matching.
+
+    Args:
+        updates (list[dict[str, Any]]): Incoming update items.
+        lookup_key (str): Identifier key used to match items (e.g., ``task_key``).
+
+    Returns:
+        tuple[list[tuple[str, dict]], list[tuple[Pattern, dict, str]]]:
+            - literal updates as ``(key, payload)``
+            - regex updates as ``(compiled_pattern, payload, pattern_str)``
+    """
+    literal_updates: list[tuple[str, dict[str, Any]]] = []
+    regex_updates: list[tuple[re.Pattern[str], dict[str, Any], str]] = []
+    for upd in copy.deepcopy(updates):
+        key_value = upd.pop(lookup_key)
+        if isinstance(key_value, str) and key_value.startswith("re:"):
+            pattern = key_value[3:]
+            try:
+                compiled = re.compile(pattern)
+            except re.error:
+                log.debug(f"Invalid regex pattern for {lookup_key}: {pattern}")
```

Comment:
This should throw.. We want a predictable output, not just skipping if a failure happens